html/dom/reflection.js

```diff
@@ -522,24 +600,68 @@ ReflectionTests.typeMap = {
             " " + binaryString + " foo ", undefined, 1.5, "5%", "+100", ".5", true, false,
             {"test": 6}, NaN, +Infinity, -Infinity, "\0",
             {toString:function() {return 2;}, valueOf: null},
-            {valueOf:function() {return 3;}}],
-        "domExpected": [minInt - 1, minInt, -36, -1, 0, 1, maxInt,
-                        maxInt + 1, maxUnsigned, maxUnsigned + 1, null,
-                        // Leading whitespace tests
-                        7, null, 7, 7, null, null,
-                        7, 7, null, null, null, null,
-                        null, null, null, null, null, null,
-                        null, null, null, null, null, null,
-                        null,
-                        // End leading whitespace tests
-                        null, null, 1.5, 5, 100, 0.5, null, null,
-                        null, null, null, null, null,
-                        2, 3],
-        // I checked that ES ToString is well-defined for all of these (I
-        // think).  Yes, String(-0) == "0".
+            {valueOf:function() {return 3;}, toString: null}],
+        "domExpected": function (val) {
+            var parsed = ReflectionTests.parseFloat(String(val));
+            if (parsed === false) {
+                return null;
+            }
+            return parsed;
+        },
+        "idlTests":       [ -10000000000,   -1,  -0,   0,   1,   10000000000],
+        "idlDomExpected": function (val) {
+            return Number(val).toFixed();
+        },
+        "idlIdlExpected": function (val) {
+            // This is a bit heavy-weight but hopefully will give values
+            // that compare "better" (without introducing some tolerance)
+            // when the test cases are expanded with more values.
+            return ReflectionTests.parseFloat(Number(val).toFixed());
+        }
+    },
+    /**
+     * Reflected IDL attribute of type double, limited to only positive values,
+     * are similar to the previous case with the following exceptions:
+     *
+     *  - on getting, if the parsed value is not greater than 0, then return
+     *    the default value
+     *  - on setting, if the value is not greater than 0, then return (leaving)
+     *    the attribute to its previous value.
+     */
+    "limited double": {
+        "jsType": "number",
+        "defaultVal": 0.0,
+        "domTests": [minInt - 1, minInt, -36, -1, 0, 1, maxInt,
+            maxInt + 1, maxUnsigned, maxUnsigned + 1, "",
+            "\u00097", "\u000B7", "\u000C7", "\u00207", "\u00A07", "\uFEFF7",
```

Comment:
In working on a patch for https://webkit.org/b/268542, I find it’d be useful to test that a U+000B isn’t handled as allowed whitespace even when the U+000B is other than the first code point in the value.
```suggestion
            "\u00097", "\u000B7", "\u000C7", "\u00207", "\u00A07", "\uFEFF7",
            "\t\u000B7", "\n\u000B7","\f\u000B7", "\r\u000B7", "\x20\u000B7",
```

Reply:
Ah interesting. This should probably be tested for all numeric values (all the longs) as well.

How should I amend the PR? Amend the appropriate commit (rebase with fixups) or pushing new commits? What are the rules here? (I don't think I saw this discussed in the docs)

Reply:
> How should I amend the PR? Amend the appropriate commit (rebase with fixups) or pushing new commits? What are the rules here?

I don’t think there’s any strict policy. So it’s fine to amend and force-push, if that’s your normal workflow elsewhere. I guess it’s also fine to push new commits, if you want to be able to roll back any changes or whatever.

As far as the reviewer workflow of reviewing locally in the shell from the checked-out PR branch, I guess there it’s easier to review when changes are amended and force-pushed rather than just pushed as new commits.

As far as the reviewer workflow of reviewing from the GitHub UI, I’m not sure how most other people review PRs with the GitHub UI — but personally, I never review from the **Commits** tab; instead I always review from the **Files changed** tab. So for the GitHub UI case for me at least, it doesn’t really matter whether somebody has pushed new commits instead of amending and force-pushing. I suspect that probably holds true for most other people as well.

Reply:
Added those tests to all numeric types.